### PR TITLE
Write ValueSetComponentRules

### PR DESCRIPTION
**Pull request by ngfreiter**
This adds writing of `ValueSetComponentRule`s, which can either be `ValueSetConceptComponentRule`s or `ValueSetFilterComponentRule`s.

I think @mint-thompson originally implemented a lot of this code on the SUSHI side, so I think it makes sense for them to be one of the reviewers on this, to help ensure I wasn't missing any cases to test.

My manual method for testing this was to take all of the rules tested in the tests and run them with SUSHI, and verify that there were no errors, and the results were as expected.

### CODE REVIEW

**Comment by ngfreiter** [comment]
Please excuse my shameful git indiscretion.

File: src/exportable/ExportableValueSetComponentRule.ts
Comment (cmoesel) [reply src/exportable/ExportableValueSetComponentRule.ts:17]: Here are my thoughts:
1. We should _sometimes_ use `includes`.  I prefer `* bar#foo "Foo Bar"` without the `includes`.  But if a rule has the word `from` in it (like the filter rules), then I think it might read better _with_ `include`. This also makes sense because there can be _many_ individual concepts, so 50+ `include`s would get pretty repetitive -- but there aren't many filters, so it's reasonable to have 2 or 3 `include`s in that case.
2. This is trickier if we want to preserve the structure of a `ValueSet` round-trip.  Personally, I like having a rule per `system#code` (vs `#code1 and #code2 and #code3 from system`).  I believe that in this case, SUSHI would bundle all the codes from the same system into a single component when it exports the `ValueSet`.  BUT... we could get a `ValueSet` that spread codes from the same system across multiple components.  In that case, should we use the `#code1 and #code2 from system` approach to keep the components distinct in the FSH?  I can't remember if SUSHI would retain that distinction or still just merge them into a single component for the system.
3. I'm not sure of the answer on this one.  I can see arguments for each of these.

**Review by cmoesel — COMMENTED** [review]
This PR seems sound, but raises some questions for me that I think would benefit from discussion.  See my individual comment.

File: src/exportable/ExportableValueSetComponentRule.ts
Comment (ngfreiter) [reply src/exportable/ExportableValueSetComponentRule.ts:17]: Yeah all good questions that I wasn't sure I was making the right calls on.
1. I personally also prefer less `include` to more, *but* the reason I always included `include` was because that is the convention that seems to be adopted by the [spec](http://build.fhir.org/ig/HL7/fhir-shorthand/branches/master/reference.html#defining-value-sets)
2. I agree I like the `system#code` approach. Especially if these codes have `display` values. The other problem with doing `and` is it just makes things harder on us. Because there is definitely a point when you have too many codes from one single system to put them all on one line, see this [VS](https://github.com/HL7/fhir-mCODE-ig/blob/master/input/fsh/VS_CancerCondition.fsh) from mCODE as an example.
3. In my head to follow the pattern of profiles/extensions, the extractors/toFSH functions should do the simplest possible valid things. And I _think_ each component can be mapped to one rule, and each rule could then be `toFSH`ed. So to me, combining multiple rules (if we decide that best practice is to do that) seems an `optimizer` function.

**Review by cmoesel — APPROVED** [review]
OK.  I'm comfortable approving this as-is, knowing that we may choose to refine the output using optimizers in the future.

**Comment by cmoesel** [comment]
@mint-thompson or @jafeltra -- can we get a second review on this?

**Comment by cmoesel** [comment]
@ngfreiter - can you sync w/ master?

**Review by jafeltra — APPROVED** [review]
This makes sense to me!

**Pull request merged**
